docs/training/training-repo.md

```diff
@@ -0,0 +1,77 @@
+# Modify Repository Proposal: Training
+
+## Summary
+
+This document proposes taking the existing [instructlab/training](https://github.com/instructlab/training) and transforming it into a library that will be consumed
+by other clients such as the CLI and what is currently the `main_ds.py` file
+that launches a full fine-tune.
+
+## Background
+
+Today, we have training implemented for different use-cases
+existing in different repos. LoRA/QLoRA training currently lives
+in the [`instructlab`](https://github.com/instructlab/instructlab)
+CLI repo and contains logic for both Linux and MacOS training.
+Each of these then brings additional logic for platform-specific accelerators.
+For example NVIDIA CUDA and Intel Habana on Linux, and MLX and MPS on MacOS.
+
+On the other hand, we have logic that exists today for full fine-tuning
+in the [instructlab/training](https://github.com/instructlab/training) repo. Today, this implementation is CUDA-specific
+as it relies on NCCL for handling communication across various
+NVIDIA GPUs.
+We want to bring support for alternative accelerators such as Intel Gaudi to this repo as well, without rewriting much of the logic.
+
+We need to have a library which provides a common interface
+for training, allowing all of the hard logic to be concentrated
+in a single point.
+
+The [instructlab/training](https://github.com/instructlab/training) repo should then
+become the home for the overall training library.
+
+This library will then contain a "simple" interface for all
+other clients to pull from and use without much needing to change
+on their side other than how they intend to use it.
+
+## Maintainers
+
+The maintainers should be the folks who currently work on training.
+Namely:
+
+- [Aldo Pareja](https://github.com/aldopareja)
```

Comment:
I'd like to be added to this as the community train rep currently. Or @alimaredia. If we are planning on moving that code to the library, I would appreciate a part in that :pray:

Reply:
The `sdg` and `eval` repos bootstrapped access by copying the existing `Backend Maintainers` team. That's an alternative that could be used here.